src/app/Recordings/Filters/DurationFilter.tsx

```diff
@@ -38,37 +43,50 @@ export const DurationFilter: React.FC<DurationFilterProps> = ({
     [onContinuousDurationSelect],
   );
 
+  const handleSubmit = React.useCallback(() => onDurationInput(duration), [onDurationInput, duration]);
+
   const handleEnterKey = React.useCallback(
     (e) => {
-      if (e.key && e.key !== 'Enter') {
-        return;
+      if (e.key && e.key === 'Enter') {
+        handleSubmit();
       }
-      onDurationInput(duration);
     },
-    [onDurationInput, duration],
+    [handleSubmit],
   );
 
   return (
-    <Flex>
-      <FlexItem flex={{ default: 'flex_1' }}>
-        <TextInput
-          type="number"
-          value={duration}
-          id="duration-input"
-          aria-label="duration filter"
-          onChange={(e, value) => setDuration(Number(value))}
-          min="0"
-          onKeyDown={handleEnterKey}
-        />
-      </FlexItem>
-      <FlexItem alignSelf={{ default: 'alignSelfCenter' }}>
-        <Checkbox
-          className="duration-filter__continuous-checkbox"
-          label="Continuous"
-          id="continuous-checkbox"
-          isChecked={isContinuous}
-          onChange={handleContinuousCheckBoxChange}
-        />
+    <Flex spaceItems={{ default: 'spaceItemsSm' }}>
+      <Flex direction={{ default: 'column' }}>
+        <FlexItem>
+          <TextInput
+            type="number"
+            value={duration}
+            id="duration-input"
```

Comment:
This seems like a good time to enhance this duration picker input. Having to match an exact duration is pretty limiting, especially since the UI only allows for granularity of seconds. "Duration greater than" and "Duration less than" with a number spinner and unit dropdown would be nice, so the user can quickly do something like "filter for any recording at least 5 minutes long"

Reply:
Oh nice, great idea! How about this design below? The `Duration` filter is displayed as a range selection. We can do the same for `Datetime` filter (i.e. `StartedBeforeDate` and `StartedAfterDate`).

![image](https://github.com/user-attachments/assets/bf4d082a-4681-4ccb-9563-056f445767bd)

The latest commit here should have this implemention.

Reply:
Looks pretty good to me. I think adding an "unlimited" or "infinite" option to either end of the range would be good, so that it's quick to express the same "start time before X" as "start time from 'infinite past' to X", without forcing the user to pick the Unix Epoch manually as their infinite past or something like that. Maybe that just exists as a checkbox? Not sure. Or maybe the existing "start time before/after" filters should just stay as they are for that purpose?

Reply:
I guess we can follow the same pattern here with the Duration by considering "unset"/"undefined" input as `Infinity`. Something like below could mean `From 2024-09-05T14:50:23.940Z onwards`?

![image](https://github.com/user-attachments/assets/9c024d09-0697-4299-87ad-06878901940c)

Reply:
This latest commit should have the datetime range selection now for `StarTime` field of a recording :D

![image](https://github.com/user-attachments/assets/9d51f589-5061-4f5b-aae9-1f99da10fbe4)